services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py

```diff
@@ -352,32 +354,46 @@ async def _run_trigger_observation_queue_task(self) -> None:
+            dynamic_sidecar_settings: DynamicSidecarSettings = (
+                self.app.state.settings.DYNAMIC_SERVICES.DYNAMIC_SIDECAR
+            )
 
-            # NOTE: ANE: not very readable should be refactored
             if (
                 scheduler_data.dynamic_sidecar.status.current
                 == DynamicSidecarStatus.FAILING
             ):
-                # Nothing will be done if there is an error while interacting
-                # with the sidecar.
-                # It makes no sense to continuously occupy resources or create
-                # issues due to high request to components like the `docker damon`
-                # and the `storage service`.
-
-                # TODO:
-
-                # MIGHT be a better alternative?
-                # - if node was deleted! (you know because in theory you call
-                # remove_node without saving ths state) just cleanup sidecars
-                # and remove from observation cycle [allows users to delete nodes and
-                # still release resources]
-
-                # FOR sure we want this
-                # - if the sidecar is missing (remove the proxy if it exists)
-                # and then remove from observation cycle [allows user to manually
-                # remove failing sidecars and the entire thing to still work afterwards]
-
-                # - cleanup: after removing?
+                # potential use-cases:
+                # 1. service failed on start -> it can be removed safely
+                # 2. service must be deleted -> it can be removed safely
+                # 3. service started and failed while running (either
+                #   dy-sidecar, dy-proxy, or containers) -> it cannot be removed safely
+                # 4. service started, and failed on closing -> it cannot be removed safely
+
+                failed_while_saving_state_and_outputs = (
+                    scheduler_data.dynamic_sidecar.service_removal_state.can_save
+                    and scheduler_data.dynamic_sidecar.were_containers_created
+                )
+                if failed_while_saving_state_and_outputs:
+                    # use-cases: 3, 4
+                    # Since user data is important and must be saved, take no further
+                    # action and wait for manual intervention from support.
+                    return
+
+                # use-cases: 1, 2
+                # Cleanup all resources related to the dynamic-sidecar.
+                await remove_dynamic_sidecar_stack(
+                    scheduler_data.node_uuid, dynamic_sidecar_settings
+                )
+                await remove_dynamic_sidecar_network(
+                    scheduler_data.dynamic_sidecar_network_name
+                )
+                await self.finish_service_removal(scheduler_data.node_uuid)
+
+                logger.warning(
+                    "cleaned up %s service after error without saving "
```

Comment:
what I meant here was that:
1. if can_save = False -> then it's actually not even an error, it's the user that wants to delete a node without saving anything
2. if were_containers_created then it's due to an error

Reply:
So you only end up in here, only if there was an error while interacting with the dy-sidecar. Considering this then:
1. still an error and can_save can either be False or None. Normally, you save the state if it is set True. 
2. correct

Reply:
Sorry I don't understand the rationale with 1. why is this still an error?
If I delete a node you will receive can_save=False and it's not an error. so why will we see this message?

Reply:
What I wanted to stata above is you only hit these checks if an error was detected (look at lines 361 to 364).
Normally, when deleting a node, you do not end up here. It goes through the normal removal process. In this situation the message will not be displayed.